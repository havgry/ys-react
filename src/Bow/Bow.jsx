import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import '../base.css'
import '@youseedk/dna/css/elements/ys-bow.css'

const Bow = ({
  variant,
  className,
  ...props
}) => {
  const wrapperProps = {
    className: classNames(
      'ys-bow',
      variant && `ys-bow--${variant}`,
      className,
    ),
    ...props,
  }

  return (
    <div
      {...wrapperProps}
    />
  )
}

Bow.propTypes = {
  variant: PropTypes.oneOf(['midnight-green']),
  className: PropTypes.string,
}

Bow.defaultProps = {
  variant: null,
  className: '',
}

export default Bow
