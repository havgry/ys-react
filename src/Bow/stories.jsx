import React from 'react'
import { storiesOf } from '@storybook/react'

import { text, select } from '@storybook/addon-knobs'

import Bow from './Bow'
import '@youseedk/dna/css/elements/ys-bow.css'

const variants = {
  default: null,
  'midnight-green': 'midnight-green',
}

storiesOf('Bow', module)
  .addParameters({ jest: ['Bow'] })
  .add('Default', () => (
    <Bow
      variant={select('variant', variants, variants.default, 'Visuals')}
      className={text('className', '', 'Misc')}
    />
  ))
