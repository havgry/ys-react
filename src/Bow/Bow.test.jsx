import React from 'react'
import { shallow, mount } from 'enzyme'

import Bow from './Bow'

describe('<Bow>', () => {
  describe('Props', () => {
    it('Has correct defaultProps', () => {
      const component = mount(
        <Bow />
      )

      const {
        variant,
        className,
      } = component.props()

      expect(variant).toBe(null)
      expect(className).toBe('')
    })
  })

  describe('Rendering', () => {
    it('Outputs elements with the correct class names', () => {
      expect(
        shallow(
          <Bow />,
        )
          .find('div.ys-bow')
          .exists(),
      ).toBe(true)
    })

    it('Custom `className` is being applied to the container', () => {
      expect(
        shallow(
          <Bow
            className="gecko"
          />,
        )
          .find('.ys-bow.gecko')
          .exists(),
      ).toBe(true)
    })

    it('Forwards other generic props to container element', () => {
      expect(
        shallow(
          <Bow
            id="gastropod"
          />,
        ).props().id,
      ).toBe('gastropod')
    })
  })

  describe('Snapshots', () => {
    it('With custom `className`', () => {
      expect(
        shallow(
          <Bow
            className="donkey"
          />,
        ),
      ).toMatchSnapshot()
    })

    it('With `midnight-green` variant', () => {
      expect(
        shallow(
          <Bow
            variant="midnight-green"
          />,
        ),
      ).toMatchSnapshot()
    })
  })
})
