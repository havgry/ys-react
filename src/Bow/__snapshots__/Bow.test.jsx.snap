// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bow> Snapshots With \`midnight-green\` variant 1`] = `
<div
  className="ys-bow ys-bow--midnight-green"
/>
`;

exports[`<Bow> Snapshots With custom \`className\` 1`] = `
<div
  className="ys-bow donkey"
/>
`;
